### PR TITLE
Decode class and method names in the traces of the linker.

### PR DESCRIPTION
**Pull request by sjrd**
This way, we show meaningful names to the developer, instead of
internal encodings.

### CODE REVIEW

**Comment by sjrd** [comment]
Review by @nicolasstucki

File: tools/shared/src/main/scala/org/scalajs/core/tools/optimizer/Analysis.scala [hunk 1]
```diff
@@ -217,8 +242,8 @@ object Analysis {
         log(level, "involving instantiated classes:")
         indented {
           for (classInfo <- involvedClasses.result().distinct) {
-            log(level, s"$classInfo")
-            if (onlyOnce(classInfo))
+            log(level, classInfo.displayName)
+            if (onlyOnce(Level.Debug, classInfo))
               logCallStackImpl(Level.Debug,
                   classInfo.instantiatedFrom.headOption, verb = "instantiated")
```
Comment (nicolasstucki) [inline tools/shared/src/main/scala/org/scalajs/core/tools/optimizer/Analysis.scala]: Missing `if` braces.

**Comment by nicolasstucki** [comment]
Would it be possible to create a couple of simple tests?

**Comment by nicolasstucki** [comment]
Thats all, otherwise LGTM.

**Comment by sjrd** [comment]
> Would it be possible to create a couple of simple tests?

In a sense, we *have* simple tests, but they're not automated because we don't have the infrastructure for that. You can *see* the output in the logs for the linking of `noIrCheckTest`, pasted here for convenience:
```
RUNNING timeout -k 5 25m sbt ++2.10.2 noIrCheckTest/test
Java HotSpot(TM) 64-Bit Server VM warning: ignoring option MaxPermSize=512M; support was removed in 8.0
[info] Loading project definition from /localhome/jenkins/a/workspace/scalajs-task-worker/project/project
[info] Loading project definition from /localhome/jenkins/a/workspace/scalajs-task-worker/project
[info] Set current project to Scala.js (in build file:/localhome/jenkins/a/workspace/scalajs-task-worker/)
[info] Setting version to 2.10.2
[info] Reapplying settings...
[info] Set current project to Scala.js (in build file:/localhome/jenkins/a/workspace/scalajs-task-worker/)
[info] Updating {file:/localhome/jenkins/a/workspace/scalajs-task-worker/}noIrCheckTest...
...
[info] Compiling 2 Scala sources to /localhome/jenkins/a/workspace/scalajs-task-worker/no-ir-check-test/target/scala-2.10/test-classes...
[warn] Referring to non-existent class scala.collection.parallel.Splitter$
[warn]   called from org.scalajs.testsuite.noircheck.RhinoLinkFailureTest$$anonfun$1.apply$mcV$sp()scala.Unit
[warn]   called from org.scalajs.jasminetest.JasmineTask.execute(sbt.testing.EventHandler,[sbt.testing.Logger,scala.Function1)scala.Unit
[warn]   called from org.scalajs.testinterface.internal.Slave$$anonfun$1.apply$mcV$sp()scala.Unit
[warn]   called from org.scalajs.jasminetest.JasmineTask.execute(sbt.testing.EventHandler,[sbt.testing.Logger,scala.Function1)scala.Unit
[warn]     (already seen, not repeating call stack)
[warn] involving instantiated classes:
[warn]   org.scalajs.testsuite.noircheck.RhinoLinkFailureTest$$anonfun$1
[warn]   org.scalajs.jasminetest.JasmineTask
[warn]   org.scalajs.testinterface.internal.Slave$$anonfun$1
[warn] Referring to non-existent method scala.collection.parallel.Splitter$.empty()scala.collection.parallel.Splitter
[warn]   called from org.scalajs.testsuite.noircheck.RhinoLinkFailureTest$$anonfun$1.apply$mcV$sp()scala.Unit
[warn]   called from org.scalajs.jasminetest.JasmineTask.execute(sbt.testing.EventHandler,[sbt.testing.Logger,scala.Function1)scala.Unit
[warn]   called from org.scalajs.testinterface.internal.Slave$$anonfun$1.apply$mcV$sp()scala.Unit
[warn]   called from org.scalajs.jasminetest.JasmineTask.execute(sbt.testing.EventHandler,[sbt.testing.Logger,scala.Function1)scala.Unit
[warn]     (already seen, not repeating call stack)
[warn] involving instantiated classes:
[warn]   org.scalajs.testsuite.noircheck.RhinoLinkFailureTest$$anonfun$1
[warn]   org.scalajs.jasminetest.JasmineTask
[warn]   org.scalajs.testinterface.internal.Slave$$anonfun$1
[warn] Referring to non-existent method scala.concurrent.forkjoin.ForkJoinWorkerThread.<init>(scala.concurrent.forkjoin.ForkJoinPool)
[warn]   called from org.scalajs.testsuite.noircheck.DummyParentsTest$$anonfun$1$$anonfun$apply$mcV$sp$1$DummyFJWorkerThread$1.<init>(org.scalajs.testsuite.noircheck.DummyParentsTest$$anonfun$1$$anonfun$apply$mcV$sp$1)
[warn]   called from org.scalajs.testsuite.noircheck.DummyParentsTest$$anonfun$1$$anonfun$apply$mcV$sp$1.apply$mcV$sp()scala.Unit
[warn]   called from org.scalajs.jasminetest.JasmineTask.execute(sbt.testing.EventHandler,[sbt.testing.Logger,scala.Function1)scala.Unit
[warn]   called from org.scalajs.testinterface.internal.Slave$$anonfun$1.apply$mcV$sp()scala.Unit
[warn]   called from org.scalajs.jasminetest.JasmineTask.execute(sbt.testing.EventHandler,[sbt.testing.Logger,scala.Function1)scala.Unit
[warn]     (already seen, not repeating call stack)
[warn] involving instantiated classes:
[warn]   org.scalajs.testsuite.noircheck.DummyParentsTest$$anonfun$1$$anonfun$apply$mcV$sp$1
[warn]   org.scalajs.jasminetest.JasmineTask
[warn]   org.scalajs.testinterface.internal.Slave$$anonfun$1
[info] 
[info] Rhino Linking
[info]   + should throw an exception if it fails loading a class
[info] 
[info] Total for suite Rhino Linking
[info] 1 specs, 0 failure
[info] 
[info] 
[info] Linking Stages
[info]   + should provide dummy parents if required
[info] 
[info] Total for suite Linking Stages
[info] 0 specs, 0 failure
[info] 
[info] Passed: Total 2, Failed 0, Errors 0, Passed 2
[success] Total time: 12 s, completed Jan 7, 2016 5:54:25 PM
```

**Comment by sjrd** [comment]
Updated.

**Comment by nicolasstucki** [comment]
Ok, then LGTM.

**Pull request merged**